src/huggingface_hub/keras_mixin.py

```diff
@@ -149,13 +159,13 @@ def save_pretrained_keras(
     config: Optional[Dict[str, Any]] = None,
     include_optimizer: Optional[bool] = False,
     plot_model: Optional[bool] = True,
+    tags: Optional[Union[list, str]] = None,
```

Comment:
The issue with this approach is that for people that were using the `**model_save_kwargs`, this will break as the function would be expecting task_name now. 

@adrinjalali do you have any suggestions here? The goal is to deprecate `task_name` without making a breaking change, but since there is `**model_save_kwargs` it might be a bit more complex.

Reply:
re backward compatibility, since these are positional args, you need to change the order between `tags` and `task_name` for people who were passing things positionally.

re `**model_save_kwargs`, after deprecation cycle is over, in this function we can have something like: ```if 'task_name' in model_save_kwargs: raise TypeError('You should be using `tags` instead of `task_name`')``` since `task_name` is not an arg to `save_model`.

Reply:
That said, I think backward compatibility constraints that we want to impose on `huggingface_hub` core parts don't necessarily apply here since the users here are mostly end users rather than library developers.

Reply:
> re backward compatibility, since these are positional args, you need to change the order between tags and task_name for people who were passing things positionally.

With this change `tags` support both a list and an individual string, so this would still support positional params just fine. That is, the value os `task_name` is a valid value in `tags` and it has the same effect. 

Given your idea, I think we could entirely remove `task_name` from the param in this PR? We could just get it (if it is there) from `model_save_kwargs`. WDYT?

Reply:
that's true. Sounds good to me, we manually raise a warning now, and later a `TypeError`.

Reply:
You could delete `task_name` from  `model_save_kwargs` after checking if it's there (or just pop it)